### DISCUSSION TITLE
Which framework to use graphql-java or graphql-java-kickstart

### DISCUSSION THREAD

**Discussion by VishnuGupta16** [graphql-java-kickstart/graphql-spring-boot#562] [Q&A]
Hi All,

I am currently exploring graphql implementation for my project, I am using Spring Boot 2.1.8-RELEASE.

I am confused as to which one to use between graphql-java or graphql-java-kickstart. 

I am looking for regular updates for new features and bug fixes.

**Comment by rmuller** [comment]
`graphql-java` is the core implementation of the GraphQL specification for Java. Almost all Java based tools and (GraphQL) frameworks use this library, including `graphql-java-kickstart`.
So you do not need to make a choice here :)

**Comment by wiedehoeft** [comment]
We are just making the same explorations; having a Spring-Boot project and determine which version to use.
Graphql-Java or the kickstart project.
After we have seen, which configuration is needed for a plain Graphql-Java project, we decided to use the kickstarte libraries.
I would suggest to explore the graphql-java documentation first https://www.graphql-java.com/documentation/v16/
and then explore documentation for this project: https://www.graphql-java-kickstart.com/spring-boot/

The start with the kickstart project was quite simple, but has some pitfalls, too.
Good:
Write a schema file and a GraphqlQueryResolver/GraphqlMutationResolver and you are good to go.

Bad:
The documentation is sometimes lacking important details, e.g. that a SchemaParser seems just to be injectable (as I recognized this morning after reading kickstart source code).
When you have an existing Spring-Boot project you must currently careful about classpath resolution. See this discussion:
https://github.com/graphql-java-kickstart/graphql-spring-boot/discussions/607

I am appreciate to hear from your explorations, when starting with this project :)
